helix-core/src/main/java/org/apache/helix/task/WorkflowDispatcher.java

```diff
@@ -174,6 +174,16 @@ public void updateWorkflowStatus(String workflow, WorkflowConfig workflowCfg,
       }
       return;
     }
+
+    // Step 6: handle workflow that should go IN_PROGRESS state after is has been resumed
+    // This block is added to make sure if workflow/queue context state becomes IN_PROGRESS for the
+    // case where A Queue which has been stopped before is just resumed and queue does not contain
+    // any job that needs to be run.
+    if (targetState.equals(TargetState.START)
+        && workflowCtx.getWorkflowState().equals(TaskState.STOPPED)) {
```

Comment:
`workflowCtx.getWorkflowState()` is not going to be null right? If it could return null, we should do a reverse check.

Also, for enums, == would be the standard for comparison vs using equals().

Reply:
No it can't. If it is null, the method will return NOT_STARTED:
```
  public TaskState getWorkflowState() {
    String state = _record.getSimpleField(WorkflowContextProperties.STATE.name());
    if (state == null) {
      return TaskState.NOT_STARTED;
    }

    return TaskState.valueOf(state);
  }
```

Reply:
Thanks for the review. I will change it to ==